src/app/shared/psat-sankey/psat-sankey.component.ts

```diff
@@ -307,15 +305,38 @@ export class PsatSankeyComponent implements OnInit {
         value: (usefulOutput / results.motor_power) * 100,
         x: .85,
         y: .65,
+        source: this.drive > 0 ? 7 : 5,
+        target: [],
+        isConnector: false,
         nodeColor: this.nodeArrowColor,
         id: 'usefulOutput'
       }
     );
 
-    
     return nodes;
   }
 
+  buildLinks(nodes, links) {
+    this.connectingLinkPaths.push(0);
+    
+    for (let i = 0; i < nodes.length; i++) {
+      if (nodes[i].isConnector) {
+          this.connectingNodes.push(i); 
+          if (i !== 0 && i-1 !== 0) {
```

Comment:
These connectingLinkPaths may be better off set from a isConnectingLink property on the node.